ocrd/ocrd/cli/workspace.py

```diff
@@ -159,6 +167,54 @@ def workspace_add_file(ctx, file_grp, file_id, mimetype, page_id, force, local_f
     )
     workspace.save_mets()
 
+# ----------------------------------------------------------------------
+# ocrd workspace add-group
+# ----------------------------------------------------------------------
+
+@workspace_cli.command('add-group')
+@click.option('-m', '--mimetype', help="Media type of the file", required=True)
+@click.option('-f', '--force', help="Continue even if mets:file same ID already exists", default=False, is_flag=True)
+@click.argument('file_grp_dir', type=click.Path(dir_okay=True, file_okay=False, readable=True, resolve_path=True), required=True)
+@pass_workspace
+def workspace_add_file_group(ctx, mimetype, force, file_grp_dir):
+    """
+    Add a directory of files as a fileGrp to METS.
+
+    Convention:
+        FILEGRP_DIR must reside within the workspacee
+        fileGrp ID = last folder name component
+        page ID = filename sans extension
+        file ID = sanitized concatenation of fileGrp ID and page ID
+    """
+    workspace = Workspace(ctx.resolver, directory=ctx.directory, mets_basename=ctx.mets_basename, automatic_backup=ctx.automatic_backup)
+
+    file_grp_dir = Path(file_grp_dir).resolve()
+
+    ws_dir = Path(ctx.directory).resolve()
+    print(file_grp_dir)
+
+    # XXX will raise ValueError if file_grp_dir isn't relative to ctx.directory
+    file_grp_dir.relative_to(ws_dir)
+    file_grp = clean_id(file_grp_dir.name)
+
+    for f in file_grp_dir.iterdir():
+        page_id = f.name
+        if f.suffix:
+            page_id = page_id[0:-len(f.suffix)]
+        page_id = clean_id(page_id)
+        file_id = clean_id('_'.join([file_grp, page_id]))
+        local_filename = str(f.relative_to(ws_dir))
+        workspace.mets.add_file(
```

Comment:
Another point (from #446): If the directory is large, then we should not use the regular `add_file()` here, which tries to find the ID and then adds it to `workspace.mets._file_by_id` dict: this creates considerable memory overhead and slows down. 

Instead, here we should check in bulk that the file_ids we want to add do not yet exist, and then add to the workspace without the dict mechanism.

Reply:
Probably best to fix #446 first then.

Reply:
Agreed. But still, avoiding the `self.find_files(ID=ID)` test in each iteration would make this more efficient. Perhaps by adding an option `ignore=False` complementing `force=False`?

Reply:
`ignore` is a good idea, implemented in 35289a13